agent/functions/put_file_test.go

```diff
@@ -0,0 +1,45 @@
+package functions_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github/clover0/github-issue-agent/functions"
+)
+
+func TestPutFile(t *testing.T) {
+	t.Run("successfully writes content to file", func(t *testing.T) {
+		outputPath := "testdata/testfile.txt"
+		content := "Hello, World!"
+
```

Comment:
from CodeQualityAnalyst
deferでファイルを削除する前に、ファイルが存在するかどうかを確認することをお勧めします。

```suggestion
if _, err := os.Stat(outputPath); err == nil {
	defer os.Remove(outputPath)
}
```